sql/dd/impl/bootstrap/bootstrapper.cc

```diff
@@ -1072,6 +1072,27 @@ bool create_dd_schema(THD *thd) {
 }
 
 bool initialize_dd_properties(THD *thd) {
+  if (!opt_initialize) {
+    // Find out which ddse contains dd_properties table during restart
+    // Try rocksdb first
+    handlerton *rocksdb_ddse = ha_resolve_by_legacy_type(thd, DB_TYPE_ROCKSDB);
+    int error = HA_ERR_NO_SUCH_TABLE;
+    if (rocksdb_ddse != nullptr || rocksdb_ddse->table_exists_in_engine) {
+      error = rocksdb_ddse->table_exists_in_engine(
+          rocksdb_ddse, thd, MYSQL_SCHEMA_NAME.str,
+          dd::tables::DD_properties::instance().name().c_str());
+    }
+    if (error == HA_ERR_TABLE_EXIST) {
+      // actual ddse maybe rocksdb
```

Comment:
Regarding the atomicity of moving the table between engines, add `assert(!innodb_hton->table_exists_in_engine(the same table))`?

Reply:
innodb doesn't have table_exists_in_engine.. maybe assert mysql.idb exists?

Reply:
In this case the current code is fine